sql/core/src/main/scala/org/apache/spark/sql/execution/ui/SQLAppStatusListener.scala

```diff
@@ -77,8 +77,28 @@ class SQLAppStatusListener(
 
     val executionId = executionIdString.toLong
     val jobId = event.jobId
+    val isExecNotExists = liveExecutions.get(executionId) == null
     val exec = getOrCreateExecution(executionId)
 
+    if (isExecNotExists) {
+      val sqlStoreData =
```

Comment:
There are a few issues.

`Some(anything).nonEmpty` will always be true.

If `isExecNotExists` is true but the execution is not in the kvstore, then you'll get an exception. Not sure that can actually happen, but looking at just this code, you'd get an error in that situation.

You need something like:

```
val exec = Option(liveExecutions.get(executionId))
  .orElse {
    try {
      Some(kvstore.read(...))
    } catch {
      case NoSuchElement => None
    }
  }.getOrElse(getOrCreateExecution(executionId))
```

Reply:
Done.